### PR TITLE
fix: rework data-grid scroll into view

### PR DESCRIPTION
**Pull request by scomea**
## 📖 Description

Data-grid only invoked scroll into view when it was the scroll container itself and not as a variable height container that was taller than the page.  This change updates scroll-into-view behavior and keyboard page up/down behavior to handle all cases.

This changes the api of FASTDataGrid by adding a "page-size" attribute that allows authors to control how many items to jump over during page up/down operations.  If unset the grid uses either it's own height/grid row height or document height/grid row height depending on whether the grid itself is a scrolling container or not.

### 🎫 Issues

closes https://github.com/microsoft/fast/issues/6119

## 📑 Test Plan

Hoping to add playwright tests when that is available

### General
- [x] I have included a change request file using `$ yarn change`
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific
- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: change/@microsoft-fast-foundation-18dec58f-9866-409f-be3a-8dcaedfc8a92.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "rework scroll into view",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "stephcomeau@msn.com",
+  "dependentChangeType": "patch"
```
Comment (nicholasrice) [inline change/@microsoft-fast-foundation-18dec58f-9866-409f-be3a-8dcaedfc8a92.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

File: packages/web-components/fast-foundation/src/data-grid/data-grid.ts [hunk 2]
```diff
@@ -520,17 +507,48 @@ export class FASTDataGrid extends FASTElement {
...
+
+        pageSize = Math.max(Math.round(pageSize), 1);
+        return pageSize;
+    }
+
     private focusOnCell = (
         rowIndex: number,
         columnIndex: number,
-        scrollIntoView: boolean
+        alignment: "start" | "center" | "end" | "nearest"
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/src/data-grid/data-grid.ts:551]: What does "nearest" mean in this context? I'm not following that

File: packages/web-components/fast-foundation/src/data-grid/data-grid.ts [hunk 3]
```diff
@@ -424,92 +438,65 @@ export class FASTDataGrid extends FASTElement {
...
-                        }
-                        this.scrollTop = thisRow.offsetTop - stickyHeaderOffset;
-                        break;
-                    }
-                }
-
-                this.focusOnCell(newFocusRowIndex, this.focusColumnIndex, false);
+                console.log(this.getPageSize());
+                console.log(this.focusRowIndex);
+                console.log(this.focusRowIndex + this.getPageSize());
```
Comment (nicholasrice) [inline packages/web-components/fast-foundation/src/data-grid/data-grid.ts:486]: Nit, remove logs

Comment (scomea) [reply packages/web-components/fast-foundation/src/data-grid/data-grid.ts:479]: done

File: change/@microsoft-fast-foundation-18dec58f-9866-409f-be3a-8dcaedfc8a92.json (hunk 1 above)
Comment (scomea) [reply change/@microsoft-fast-foundation-18dec58f-9866-409f-be3a-8dcaedfc8a92.json]: done

File: packages/web-components/fast-foundation/src/data-grid/data-grid.ts (hunk 2 above)
Comment (scomea) [reply packages/web-components/fast-foundation/src/data-grid/data-grid.ts]: It's the parameters for the "block" parameter of a "scrollIntoView" call.  I changed the param in the `focusOnCell` function to be the actual system type of `ScrollLogicalPosition' to make it clearer. 
https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView
`

File: packages/web-components/fast-foundation/src/data-grid/data-grid.ts (hunk 3 above)
Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/data-grid/data-grid.ts:479]: @scomea, still seeing the console logs :)

File: packages/web-components/fast-foundation/src/data-grid/stories/data-grid.stories.ts [hunk 4]
```diff
@@ -5,67 +5,64 @@ import { FASTDataGrid, GenerateHeaderOptions } from "../data-grid.js";
 
 const storyTemplate = html<StoryArgs<FASTDataGrid>>`
     <fast-data-grid
+        style="${x => x.style}"
         :rowsData="${x => x.rowsData}"
-        ?no-tabbing="${x => x.noTabbing}"
+        no-tabbing="${x => x.noTabbing}"
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/data-grid/stories/data-grid.stories.ts:10]: this is still a boolean attribute, yes?

**Review by chrisdholt — APPROVED** [review]
Approved assuming the removal of the console logs (still seeing them)

File: packages/web-components/fast-foundation/src/data-grid/data-grid.ts (hunk 3 above)
Comment (scomea) [reply packages/web-components/fast-foundation/src/data-grid/data-grid.ts]: Oops.  Fixed.

**Review by nicholasrice — APPROVED** [review]

File: packages/web-components/fast-foundation/src/data-grid/stories/data-grid.stories.ts (hunk 4 above)
Comment (scomea) [reply packages/web-components/fast-foundation/src/data-grid/stories/data-grid.stories.ts:10]: yes, just removed the redundant "?"

**Pull request merged**